examples/play_file.py

```diff
@@ -0,0 +1,14 @@
+#Plays a file on the specified pins.
+import audio
+
+def audio_generator(file, frame):
+    ln = -1
+    while ln:
+        ln = file.readinto(frame)
+        yield frame
+
+def play_file(name, pin=None, return_pin=None):
+    #Do allocation here, as we can't do it in an interrupt.
+    frame = audio.AudioFrame()
+    with open(name) as file:
+        audio.play(audio_generator(file, frame), pin=pin, return_pin=return_pin)
```

Comment:
I guess you can now rewrite this example to create the `frame` and open the `file` at the start of `audio_generator`?  Not important though.